OpenEdXMobile/src/main/java/org/edx/mobile/util/UrlUtil.java

```diff
@@ -23,4 +29,35 @@ public static String makeAbsolute(String url, String base) {
             return null;
         }
     }
+
+    /**
+     * Builds a valid URL with the given query params.
+     *
+     * @param logger      For logging purpose.
+     * @param baseUrl     The base URL.
+     * @param queryParams The query params to add in the URL.
+     * @return URL String with query params added to it.
+     */
+    public static String buildUrlWithQueryParams(@NonNull Logger logger, @NonNull String baseUrl,
+                                                 @NonNull Map<String, String> queryParams) {
+        // TODO: Use Uri.Builder class for building a URL instead of the logic below
```

Comment:
We have done enough changes in this utility function, will be better to do this TODO within this PR.